### PR TITLE
Cordexupdate

### PR DESCRIPTION
**Pull request by paolap**
Add CORDEX query (issues #47 and #146 )
Modified flow for requests (issue #145 )
Added saving file message where appropriate (issue #140 )
All documentation and training updated
ready for new release

### CODE REVIEW

File: clef/cli.py [hunk 1]
```diff
@@ -32,7 +31,11 @@
 from .code import call_local_query, matching, write_csv, print_stats, ids_df
 from .helpers import load_vocabularies, fix_model, fix_path, get_ids
 from .esdoc import citation, write_cite
-
+#HERE possibly can delete the commented imports
+#import json
+#import pkg_resources
+#import re
```
Comment (ScottWales) [inline clef/cli.py:37]: Comments can be cleaned up

File: clef/code.py [hunk 2]
```diff
@@ -14,7 +14,11 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-
+# Most probably to delete
+#from datetime import datetime, timedelta
+#from sqlalchemy import any_, or_
+#from sqlalchemy.orm import aliased
+#from itertools import groupby
```
Comment (ScottWales) [inline clef/code.py:21]: Same here

**Review by ScottWales — APPROVED** [review]
Thanks Paola, looks good. Just a couple comments that can be cleaned up if you like. I've fixed the tests - was a dependency issue from the latest version of sqlalchemy

File: clef/code.py (hunk 2 above)
Comment (paolap) [reply clef/code.py:21]: Thanks, I probably meant to do that once I was sure everything worked and forgot!
I'll do it now!

Comment (paolap) [reply clef/code.py]: done!

**Pull request merged**